lightning/src/ln/outbound_payment.rs

```diff
@@ -1813,6 +1819,30 @@ impl OutboundPayments {
 	pub fn clear_pending_payments(&self) {
 		self.pending_outbound_payments.lock().unwrap().clear()
 	}
+
+	pub fn set_awaiting_invoice_flag(&self) {
+		let pending_outbound_payments = self.pending_outbound_payments.lock().unwrap();
+		let has_invoice_requests = pending_outbound_payments.values().any(|payment| {
+			matches!(payment, PendingOutboundPayment::AwaitingInvoice { invoice_request: Some(_), .. })
+		});
+		self.awaiting_invoice_flag.store(has_invoice_requests, Ordering::SeqCst);
+	}
+
+	pub fn get_invoice_request_awaiting_invoice(&self) -> Vec<InvoiceRequest> {
+		if !self.awaiting_invoice_flag.load(Ordering::SeqCst) {
```

Comment:
Thanks for the pointer, Matt!
I went with `SeqCst` because I wasn't entirely sure which memory ordering was the best fit, so I opted for the strictest one to play it safe. Do you have any recommendations on what would work better here?

Thank you so much for all your help and suggestion, Matt!

Reply:
Thanks a lot for the references, Matt!
I fell into the memory management rabbit hole and almost forgot I had to update my code too! 😅

I've updated the ordering to Acquire and Release in [893401e](https://github.com/lightningdevkit/rust-lightning/commit/893401e404bd3fc01f69b402d520116b5a17c50e).
Thanks a lot for the help!